types/node/test.d.ts

```diff
@@ -1050,3 +966,189 @@ declare module 'node:test' {
+        /**
+         * The duration of the test in milliseconds.
+         */
+        duration: number;
+        /**
+         * The error thrown by the test.
+         */
+        error: Error;
+    };
+    /**
+     * The test name.
+     */
+    name: string;
+    /**
+     * The nesting level of the test.
+     */
+    nesting: number;
+    /**
+     * The ordinal number of the test.
+     */
+    testNumber: number;
+    /**
+     * Present if `context.todo` is called.
+     */
+    todo?: string | boolean;
+    /**
+     * Present if `context.skip` is called.
+     */
+    skip?: string | boolean;
+    /**
+     * The path of the test file, undefined if test is not ran through a file.
+     */
+    file?: string;
+}
+interface TestPass {
+    /**
+     * Additional execution metadata.
+     */
+    details: {
+        /**
+         * The duration of the test in milliseconds.
+         */
+        duration: number;
+    };
+    /**
+     * The test name.
+     */
+    name: string;
+    /**
+     * The nesting level of the test.
+     */
+    nesting: number;
+    /**
+     * The ordinal number of the test.
+     */
+    testNumber: number;
+    /**
+     * Present if `context.todo` is called.
+     */
+    todo?: string | boolean;
+    /**
+     * Present if `context.skip` is called.
+     */
+    skip?: string | boolean;
+    /**
+     * The path of the test file, undefined if test is not ran through a file.
+     */
+    file?: string;
+}
+interface TestPlan {
+    /**
+     * The nesting level of the test.
+     */
+    nesting: number;
+    /**
+     * The number of subtests that have ran.
+     */
+    count: number;
+    /**
+     * The path of the test file, undefined if test is not ran through a file.
+     */
+    file?: string;
+}
+interface TestStart {
+    /**
+     * The test name.
+     */
+    name: string;
+    /**
+     * The nesting level of the test.
+     */
+    nesting: number;
+    /**
+     * The path of the test file, undefined if test is not ran through a file.
+     */
+    file?: string;
+}
+interface TestStderr {
+    /**
+     * The path of the test file, undefined if test is not ran through a file.
+     */
+    file?: string;
+    /**
+     * The message written to `stderr`
+     */
+    message: string;
+}
+interface TestStdout {
+    /**
+     * The path of the test file, undefined if test is not ran through a file.
+     */
+    file?: string;
+    /**
+     * The message written to `stdout`
+     */
+    message: string;
+}
+
+/**
+ * The `node:test/reporters` module exposes the builtin-reporters for `node:test`.
+ * To access it:
+ *
+ * ```js
+ * import test from 'node:test/reporters';
+ * ```
+ *
+ * This module is only available under the `node:` scheme. The following will not
+ * work:
+ *
+ * ```js
+ * import test from 'test/reporters';
+ * ```
+ * @since v19.9.0
+ * @see [source](https://github.com/nodejs/node/blob/v20.2.0/lib/test/reporters.js)
+ */
+declare module 'node:test/reporters' {
```

Comment:
This should really be in it's own file eg. `test/reporters.d.ts`

Reply:
I wanted to do that, but as it shares some types I was not sure what would be the way to do that...

Reply:
BTW, I did it this way because I saw `utils/types` is in the same file with `utils` 🙂